tests/riotboot/tests/01-run.py

```diff
@@ -13,68 +13,91 @@
-
-    # Flash to both slots and verify basic functions
-    for version in [current_app_ver + 1, current_app_ver + 2]:
-        slotnum = slotnum ^ 1
-        flash_slot(slotnum, version)
-        assert_check_slot(child, slotnum, version)
-
-    print("TEST PASSED")
+class RiotbootDevice:
+    def __init__(self):
+        self.slot_num = None
+        self.app_ver = None
+
+    def flash_current_slot_and_app_ver(self):
+        cmd = [
+            MAKE,
+            "RIOTBOOT_SKIP_COMPILE=1",
+            "riotboot/flash-slot{}".format(self.slot_num),
+            "APP_VER={}".format(self.app_ver),
+        ]
+        assert(self.slot_num is not None)
+        assert(self.app_ver is not None)
+        assert subprocess.call(cmd) == 0
+
+    def get_current_slot_and_app_ver(self, child):
+        # Ask for current slot, should be 0 or 1
+        child.expect_exact('>')
+        child.sendline("curslotnr")
+        child.expect(r"Current slot=([0-1])")
+        self.slot_num = int(child.match.group(1))
+
+        # Ask for current APP_VER
+        child.expect_exact('>')
+        child.sendline("curslothdr")
+        child.expect(r"Image Version: (?P<app_ver>0x[0-9a-fA-F]{8})\r\n")
+        self.app_ver = int(child.match.group("app_ver"), 16)
+
+    def check_current_slot_and_app_ver(self, child):
+        # Check if it's running on the expected slot
+        child.expect_exact('>')
+        child.sendline("curslotnr")
+        child.expect_exact("Current slot={}".format(self.slot_num))
+
+        # Ask for current slot header info and check for basic output integrity
+        child.expect_exact('>')
+        child.sendline("curslothdr")
+        # Magic number is "RIOT" (always in little endian)
+        child.expect_exact("Image magic_number: 0x544f4952")
+        # Other info is hardware/app dependent so we just check basic compliance
+        child.expect_exact("Image Version: {0:#0{1}x}".format(self.app_ver, 10))
+        child.expect(r"Image start address: 0x[0-9a-fA-F]{8}\r\n")
+        child.expect(r"Header chksum: 0x[0-9a-fA-F]{8}\r\n")
+
+        # Ask for address of slot 0
+        child.expect_exact('>')
+        child.sendline("getslotaddr 0")
+        child.expect(r"Slot 0 address=0x[0-9a-fA-F]{8}\r\n")
+
+        # Ask for data of all slots
+        child.expect_exact('>')
+        child.sendline("dumpaddrs")
+        child.expect(r"slot 0: metadata: 0x[0-9a-fA-F]{1,8} "
+                     r"image: 0x[0-9a-fA-F]{8}\r\n")
+        child.expect(r"slot 1: metadata: 0x[0-9a-fA-F]{1,8} "
+                     r"image: 0x[0-9a-fA-F]{8}\r\n")
+        child.expect_exact('>')
+
+
+def testfunc():
+    # This test requires flashing slot binaries. Some flashers like
+    # cc2538-bsl need to attach to the serial terminal in order to be
+    # able to flash. Since `run(testfunc)` spawns a terminal a different
+    # `run(testfunc)` is called after every flash in order to detach and
+    # re-attach the terminal between flashes
```

Comment:
I think this part of the comment needs to be adapted (`run(testfunc)` is no longer used).

Reply:
I'll change to `run(func)` it was meant to be generic.